Assets/Scripts/GameEvent.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "NewGameEvent", menuName = "CustomSO/Events/GameEvent")]
+public class GameEvent : ScriptableObject
+{
+
+    private List<GameEventListener> listeners = new List<GameEventListener>();
+    public void RegisterListerner(GameEventListener listener)
+    {
+        listeners.Add(listener);
+    }
+
+    public void UnregisterListener(GameEventListener listener)
+    {
+        listeners.Remove(listener);
+    }
+
+    public void Raise()
+    {
+        for (int i = listeners.Count - 1; i >= 0; --i)
+        {
+            listeners[i].RaiseEvent();
+        }
```

Comment:
This is cool.  Have you looked into Actions?  I just toyed around with it today. I can't tell right now if GameEvent and Actions are any different, and they are probably implemented very similar, but I just wanted to share
```
class Bomb: MonoBehaviour {
    public static Action bombExplodedEvent;

   private void Detonate() {
       bombExplodedEvent?.invoke() // similar to .RaiseEvent()
   }
}

class HUD: MonoBehaviour {

    private void OnEnable() {
        Bomb. bombExplodedEvent += BombCardUsed() // similar to RegisterListener
    }

    private void OnDisable() {
        Bomb. bombExplodedEvent -= BombCardUsed() // similar to UnregisterListener
    }

    private void BombCardUsed() {
//        ...
    }
}